iree/tools/CMakeLists.txt

```diff
@@ -68,7 +68,19 @@ if(${IREE_BUILD_COMPILER})
     tensorflow::mlir_xla
   )
 
-  foreach(LIB ${_ALWAYSLINK_LIBS})
+  set(_ALWAYSLINK_LIBS_IREE_OPT
+    MLIRAffineOps
+    MLIRLinalgOps
+    MLIRStandardOps
+    MLIRSPIRV
+    MLIRSPIRVSerialization
+    MLIRSPIRVTransforms
+    MLIRTranslation
+    MLIROptMain
+    tensorflow::mlir_xla
+  )
+
+  foreach(LIB IN LISTS _ALWAYSLINK_LIBS_IREE_TRANSLATE _ALWAYSLINK_LIBS_IREE_OPT)
     # Aliased targets are always in-project, so we control them and can set
     # ALWAYSLINK on them directly.
     # TODO(scotttodd): add the ALWAYSLINK property to MLIR libraries elsewhere
```

Comment:
My idea here was that we could

* enumerate which MLIR (and other external) libs should be ALWAYSLINK in a different file
* set the `ALWAYSLINK` property on that list persistently for the entire project there
* treat these libraries as regular dependencies, as if they were from `iree_cc_library`, throughout the rest of the project

That way, this `foreach` wouldn't be handling duplicates, the deps wouldn't need special treatment in this file, and this CMake file would look closer to the corresponding BUILD file.

That can be future cleanup, separate from the `iree-opt` / `iree_opt_library` separation in this PR.

Reply:
I fully agree. One possibility is the approach in PR #518.